### PR TITLE
expose global settings and add unit tests to verify

### PR DESCRIPTION
**Pull request by BrandonOCasey**

### CODE REVIEW

File: test/videojs-contrib-hls.test.js [hunk 1]
```diff
@@ -1133,10 +1133,10 @@ QUnit.test('reloads out-of-date live playlists when switching variants', functio
 });
 
 QUnit.test('if withCredentials global option is used, withCredentials is set on the XHR object', function() {
-  let hlsOptions = videojs.options.hls;
+  let hlsOptions = videojs.Hls.options;
 
   this.player.dispose();
-  videojs.options.hls = {
```
Comment (dmlap) [inline test/videojs-contrib-hls.test.js]: Does this mean the old options location will be ignored?

Comment (BrandonOCasey) [inline test/videojs-contrib-hls.test.js]: It would I think I have a more backwards compatible solution, I will submit it shortly

**Comment by dmlap** [comment]
@BrandonOCasey what's the intended precedence on the options? It would be nice to have a test that enforced that (if we don't already).

**Comment by dmlap** [comment]
Otherwise, LGTM

**Comment by tenacex** [comment]
Looks good

**Comment by BrandonOCasey** [comment]
@dmlap we have tests to verify that mode and withCredentials options will work when passed in with a source, or passed in globally. I am not sure what your first question is asking.

**Pull request merged**